### PR TITLE
Issue3604 solar collectors (#3640)

### PR DESCRIPTION
**Pull request by mwetter**
* Add absorber area to Types.Area

* Creating separate data records for ASHRAE93 and EN12975 solar collectors, both extending from a partial record containing common inputs. Furthermore, a parameter eta_0 is added to the EN12975 record to distinguish from y_intercept which is used in ASHRAE93.

This replaces https://github.com/lbl-srg/modelica-buildings/pull/3639
It merges https://github.com/lbl-srg/modelica-buildings/pull/3639 into a new branch and from there to `issue3604_solarCollectors` to reduce the number of files that have change logs.

---------
This is for (the first part of) #3604

This PR includes the refactor of the solar collector models. The current changes can be summarised as follows:

- A refactor of the data record
  - Split up the existing data record into an ASHRAE93 and EN12975 record, both extending from a data record containing some general parameters (like area)
  - Remove obsolete inputs: `ATyp`, `dT_nominal`, `G_nominal`
  - Include a parameter `CTyp` and `C` which allows the user to calculate the heat capacity of the collector using the dry or total heat capacity (incl. fluid) when this data is available
  - Add a parameter `eta0` for the EN record, because this value might differ from `y_intercept` which was used before (and is used by the ASHRAE standard)
  - Rename parameters to comply with the data records of SRCC, Solar keygen, and SPF: `b0`, `b1`, `a1`, and `a2` instead of `B0`, `B1`, `C1`, and `C2`
  - Clarify the meaning of `IAMDiff` (IAM at 50°)
  - Update existing data records such that they comply with this new data record format (I didn’t update the values yet)
  - As a consequence of splitting up the data, I had to make the data record replaceable in `PartialSolarCollector.mo` and redeclare it in `ASHRAE93.mo` and `EN12975.mo`
- Remove everything related to concentrating solar collectors (model equations were derived for flat plate solar collectors)
- Remove the constraint that sets the minimum number of discretization segments (used to be 3), but left the default equal to 3
- Remove obsolete calculation of `QLos_nominal` and `QUse_nominal`
- Change the calculation of the representative volume based on the choice of `CTyp` in `PartialSolarCollector.mo`
- Remove obsolete parameter declarations and update parameter names in `ASHRAE93.mo` and `EN12975.mo`
- Update the Users Guide (e.g. adding warnings regarding pressure drops depending on the medium used and IAM values for evacuated tube collectors)
- Update the documentation (including references to equations in the EnergyPlus documentation) and revision history for all models (revision history still needs a specific date)
- Update unit tests scripts and to comply with the new variable names
- Update the position of models and connections in the examples, validation and controls package models (more structured).
- Update reference results (no differences, except name of some variables)

Still to be done before merging:

- [ ] Specify date in revision history

Still to be done is this or a future PR:

- Update the data records using the latest data sheets and sources
- Revise the implementation of the incidence angle modifier (currently only valid approximation for angles <60°)
- Allow defining rectangular solar collector fields with a number of collectors in series and parallel

### CODE REVIEW

File: Buildings/Fluid/SolarCollectors/Data/BaseClasses/Generic.mo [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  parameter Modelica.Units.SI.HeatCapacity C
+    "Dry or total thermal capacity of the solar thermal collector";
+  parameter Modelica.Units.SI.Volume V "Fluid volume ";
+  parameter Modelica.Units.SI.Mass mDry "Dry mass";
+  parameter Real mperA_flow_nominal(unit="kg/(s.m2)")
+    "Nominal mass flow rate per unit area of collector";
+  parameter Modelica.Units.SI.PressureDifference dp_nominal(displayUnit="Pa")
+    "Nominal pressure drop";
+  parameter Real b0 "1st incident angle modifier coefficient";
+  parameter Real b1 "2nd incident angle modifier coefficient";
```
Comment (mwetter) [inline Buildings/Fluid/SolarCollectors/Data/BaseClasses/Generic.mo:18]: Add missing unit attribute

File: Buildings/Fluid/SolarCollectors/Data/GenericEN12975.mo [hunk 2]
```diff
@@ -0,0 +1,29 @@
+within Buildings.Fluid.SolarCollectors.Data;
+record GenericEN12975
+  "Generic data record for EN12975 solar collector models"
+  extends Buildings.Fluid.SolarCollectors.Data.BaseClasses.Generic;
+
+  parameter Real IAMDiff
+    "Incidence angle modifier for diffuse irradiance (incidence angle of 50°)";
+  parameter Real eta0 "Optical efficiency (Maximum efficiency)";
+  parameter Real a1 "First order heat loss coefficient";
+  parameter Real a2 "Second order heat loss coefficient";
```
Comment (mwetter) [inline Buildings/Fluid/SolarCollectors/Data/GenericEN12975.mo:10]: Add missing units attributes.

**Comment by jelgerjansen** [comment]
@mwetter thanks for the review and already applied changes. I addressed your review comments in [this branch](https://github.com/jelgerjansen/modelica-buildings/tree/issue3604_solarCollectors_test) (which I forked from [buildings:issue3604_solarCollectors_test](https://github.com/lbl-srg/modelica-buildings/tree/issue3604_solarCollectors_test)).

Furthermore, I also updated some other inconsistencies:
- Update `B0` and `B1` to `b0` and `b1` in `BaseClasses/IAM.mo`.
- Added unit attribute to `IAMDiff` in `Data/GenericEN12975.mo`.
- Use predefined `Modelica.Units.SI.CoefficientOfHeatTransfer` for `slope` instead of a manually defined unit.
- Remove `min=3` attribute for `nSeg` in `BaseClasses/PartialSolarCollector.mo` (was already done for `BaseClasses/PartialParameters`).
- Specify date in revision history.

File: Buildings/Fluid/SolarCollectors/Data/GenericEN12975.mo (hunk 2 above)
Comment (jelgerjansen) [reply Buildings/Fluid/SolarCollectors/Data/GenericEN12975.mo:10]: I added a `unit` attribute, as well as a `min` attribute.

File: Buildings/Fluid/SolarCollectors/UsersGuide.mo [hunk 3]
```diff
@@ -1,110 +1,253 @@
...
+differently.
+The details of these calculations can be found in 
+<a href=\"modelica://Buildings.Fluid.SolarCollectors.BaseClasses\">
+Buildings.Fluid.SolarCollectors.BaseClasses</a>.
+Accordingly, data records for both test methods are available in 
+<a href=\"modelica://Buildings.Fluid.SolarCollectors.Data\">
+Buildings.Fluid.SolarCollectors.Data</a>.
+</p>
+<p>
+The computation of the
```
[the remark this answers is not in the document]
Comment (jelgerjansen) [reply Buildings/Fluid/SolarCollectors/UsersGuide.mo]: I presume I wanted to refer to the computation of the heat capacity of the solar collector, but this information is included in the section 'Performance data'. I therefore removed this unfinished sentence.

File: Buildings/Fluid/SolarCollectors/Data/BaseClasses/Generic.mo (hunk 1 above)
Comment (jelgerjansen) [reply Buildings/Fluid/SolarCollectors/Data/BaseClasses/Generic.mo:18]: I added a `unit` attribute, as well as a `min` and `max` attribute.

**Pull request merged**